run_infrastructure.py

```diff
@@ -0,0 +1,234 @@
+    """
+    for datastore in datastores:
+        if datastore in DOCKERFILE_DATASTORES:
+            setup_path = f"tests/integration_tests/{datastore}_setup.py"
+            os.system(
+                f'docker-compose {path} run {base_image} python {setup_path} || echo "no custom setup logic found for {datastore}"'
+            )
+
+
+def get_path_for_datastores(datastores: List[str]) -> str:
+    """
+    Returns the docker-compose file paths for the specified datastores
+    """
+    path: str = "-f docker-compose.yml"
+    for datastore in datastores:
+        os.system(f'echo "configuring infrastructure for {datastore}"')
+        if datastore in DOCKERFILE_DATASTORES:
+            # We only need to locate the docker-compose file if the datastore runs in Docker
+            path += f" -f docker-compose.integration-{datastore}.yml"
+        elif datastore not in EXTERNAL_DATASTORES:
+            # If the specified datastore is not known to us
+            os.system(f'echo "Datastore {datastore} is currently not supported"')
+
+    return path
+
+
+def _run_quickstart(
+    path: str,
+    image_name: str,
+) -> None:
+    os.system(f'echo "Running the quickstart..."')
+    os.system(f"docker-compose {path} up -d")
+    os.system(f"docker exec -it {image_name} python quickstart.py")
+
+
+def _open_shell(
+    path: str,
+    image_name: str,
+) -> None:
+    """
+    Opens a bash shell on the container at `image_name`
+    """
+    os.system(f'echo "Opening bash shell on {image_name}"')
+    os.system(f"docker-compose {path} run {image_name} /bin/bash")
+
+
+def _run_application(docker_compose_path: str) -> None:
+    """
+    Runs the application at `docker_compose_path` without detaching it from the shell
+    """
+    os.system(f'echo "Running application"')
+    os.system(f"docker-compose {docker_compose_path} up")
+
+
+def _run_tests(
+    datastores: List[str],
+    docker_compose_path: str,
+    pytest_path: str = "",
+) -> None:
+    """
+    Runs unit tests against the specified datastores
+    """
+    if pytest_path is None:
+        pytest_path = ""
+
+    path_includes_markers = "-m" in pytest_path
+    pytest_markers: str = ""
+    if not path_includes_markers:
+        # If the path manually specified already uses markers, don't add more here
+        if set(datastores) == set(DOCKERFILE_DATASTORES + EXTERNAL_DATASTORES):
+            # If all datastores have been specified use the generic `integration` flag
+            pytest_markers += "integration"
+        else:
+            # Otherwise only include the datastores provided
+            for datastore in datastores:
+                if len(pytest_markers) == 0:
+                    pytest_markers += f"integration_{datastore}"
+                else:
+                    pytest_markers += f" or integration_{datastore}"
+
+    environment_variables = ""
+    for datastore in EXTERNAL_DATASTORES:
+        if datastore in datastores:
```

Comment:
In this script you do a lot of looping through one list and seeing if that value is in another list. These lists are small, so it doesn't matter, but I would loop through `set(a).intersection(b)`.